File: BuildGenerator/src/Bll/Models/Hardware/Gpu.cs
namespace Bll.Models.Hardware;

public class Gpu : Abstractions.Hardware
{
    public double Price { get; set; }
    public string Link { get; set; } = string.Empty;
    public string Model { get; set; } = string.Empty;


    public int Length { get; set; }
    public int RequiredPowerSupplyWattage { get; set; }

    public Gpu()
    {
        
    }
    
    public Gpu(double price, string link, string model, int length, int requiredPowerSupplyWattage)
    {
        Price = price;
        Link = link;
        Model = model;
        Length = length;
        RequiredPowerSupplyWattage = requiredPowerSupplyWattage;
    }
}

File: HardwareInfoCollector/models/gpu.py
from models.hardware import *


class GPU(Hardware):
    Length: int
    RequiredPowerSupplyWattage: int

    def __init__(self, product_type: int, model: str, price: int, link: str,
                 length: int, required_power_supply_wattage: int):
        super().__init__(product_type, model, price, link)
        self.Length = length
        self.RequiredPowerSupplyWattage = required_power_supply_wattage

    def __str__(self):
        return f'{self.ProductType} {self.Model} {self.Price} {self.Link} {self.Length} ' \
               f'{self.RequiredPowerSupplyWattage}'
